### PR TITLE
KHR_materials_transmission - Make roughness affect thin surface refraction again

### PR DESCRIPTION
**Pull request by MiiBond**

### CODE REVIEW

**Comment by sebavan** [comment]
Would be great to not use branch with if when the value comes out from texture ?

**Comment by MiiBond** [comment]
@sebavan Indeed. I realized that at lunch today. I'll change it when I get a chance to work on this again. There is at least one more thing I need to do.

**Comment by MiiBond** [comment]
Okay, so this is fixed now.
However, I’m wondering if we need the volume ior VS surface ior logic at all. We can just use the thickness check to see if the material is volumetric or not.
Originally, I think I separated the two IOR's so that we could have IOR affect the reflections and refractions independently but I don't think that's really necessary.
This might change existing scenes that use refraction with a thickness of 0 though?

**Comment by sebavan** [comment]
I would prefer to avoid breaking changes but let s see what @Popov72 and @bghgary think ?

**Comment by bghgary** [comment]
I told @MiiBond to check with you, haha. From a glTF perspective, there is no change.

File: src/Materials/Node/Blocks/PBR/refractionBlock.ts [hunk 1]
```diff
@@ -233,7 +233,7 @@ export class RefractionBlock extends NodeMaterialBlock {
 
         effect.setFloat4(this._vRefractionInfosName, refractionTexture.level, 1 / indexOfRefraction, depth, this.invertRefractionY ? -1 : 1);
 
-        effect.setFloat3(this._vRefractionMicrosurfaceInfosName, refractionTexture.getSize().width, refractionTexture.lodGenerationScale, refractionTexture.lodGenerationOffset);
+        effect.setFloat4(this._vRefractionMicrosurfaceInfosName, refractionTexture.getSize().width, refractionTexture.lodGenerationScale, refractionTexture.lodGenerationOffset, indexOfRefraction);
```
Comment (Popov72) [inline src/Materials/Node/Blocks/PBR/refractionBlock.ts:236]: I think it's `1 / indexOfRefraction` instead?

Comment (MiiBond) [reply src/Materials/Node/Blocks/PBR/refractionBlock.ts]: Quite right. Thank you.

**Review by sebavan — CHANGES_REQUESTED** [review]

**Review by sebavan — APPROVED** [review]

File: src/Materials/PBR/pbrSubSurfaceConfiguration.ts [hunk 2]
```diff
@@ -353,6 +355,7 @@ export class PBRSubSurfaceConfiguration {
                     }
                 }
 
+                defines.SS_HAS_THICKNESS = this.minimumThickness !== 0.0 && this.maximumThickness !== 0.0;
```
Comment (Popov72) [inline src/Materials/PBR/pbrSubSurfaceConfiguration.ts:358]: Why not doing `defines.SS_HAS_THICKNESS = (this.maximumThickness - this.minimumThickness) !== 0.0;` instead?

I'm not sure that `this.minimumThickness !== 0.0` should be a requirement(?)

**Review by Popov72 — APPROVED** [review]

File: src/Materials/PBR/pbrSubSurfaceConfiguration.ts (hunk 2 above)
Comment (MiiBond) [reply src/Materials/PBR/pbrSubSurfaceConfiguration.ts:358]: Quite right.

**Comment by Popov72** [comment]
I think this one can be merged?

**Pull request merged**